src/org/zaproxy/zap/extension/ascanrulesAlpha/HttpOnlySite.java

```diff
@@ -0,0 +1,219 @@
+	public void raiseAlert(HttpMessage newRequest, String message) {
+		String newUri = newRequest.getRequestHeader().getURI().toString();
+		String otherInfoDetail = Constant.messages.getString(MESSAGE_PREFIX+"otherinfo." + message);
+		bingo(getRisk(), //Risk
+				Alert.CONFIDENCE_MEDIUM, //Confidence/Reliability
+				getName(), //Name
+				getDescription(), //Description
+				getBaseMsg().getRequestHeader().getURI().toString(), //Original URI
+				null, //Param
+				"", //Attack
+				Constant.messages.getString(MESSAGE_PREFIX+"otherinfo", otherInfoDetail, newUri), //OtherInfo 
+				getSolution(), //Solution
+				"", //Evidence
+				getCweId(), //CWE ID
+				getWascId(), //WASC ID
+				newRequest); //HTTPMessage
+	}
+
+	public URI constructURI(String redirect, URI oldURI) {
+		try {
+			return new URI(oldURI, redirect, true);
+		} catch (URIException err) {
+			try {
+				return new URI(oldURI, redirect, true);
+			} catch (URIException ex) {
+				return null;
+			}
+		}
+	}
+
+	@Override
+	public void scan() {
+		
+		if (getBaseMsg().getRequestHeader().isSecure()) { //Base request is HTTPS
+			if (log.isDebugEnabled()){
+				log.debug ("The original request was HTTPS, so there is not much point in looking further.");
+			}
+			return;
+		}
+
+		HttpMessage newRequest = getNewMsg();
+		try{
+			String host = newRequest.getRequestHeader().getURI().getHost();
+			String path = newRequest.getRequestHeader().getURI().getPath();
+			newRequest.getRequestHeader().setURI(new URI("https",null, host, 443, path));
+		} catch (URIException e) {
+			log.error("Error creating HTTPS URL from HTTP URL:", e);
+			return;
+		}
+
+		if (isStop()) {
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+getName()+" Stopping.");
+			}
+			return;
+		}
+
+		try {
+			int count = 0;
+			while(count < 10) {
+				if (isStop()) {
+					if (log.isDebugEnabled()) {
+						log.debug("Scanner "+getName()+" Stopping.");
+					}
+					return;
+				}
+				sendAndReceive(newRequest, false);
+				int status = newRequest.getResponseHeader().getStatusCode();
+				if(!HttpStatusCode.isRedirection(status)) {
+					break;
+				}
+				String redirect = newRequest.getResponseHeader().getHeader(HttpResponseHeader.LOCATION);
+				if(redirect == null || redirect.isEmpty()) {
+					raiseAlert(newRequest, "noredirection");
+					return;
+				}
+				URI oldURI = newRequest.getRequestHeader().getURI();
+				URI newURI = constructURI(redirect, oldURI);
+				if(newURI == null) {
+					raiseAlert(newRequest, "urinotencoded");
+					return;
+				}
+				newRequest.getRequestHeader().setURI(newURI);
+				if(!oldURI.getHost().equals(newURI.getHost())) {
+					raiseAlert(newRequest, "differenthosts");
+					return;
+				}
+				if(newRequest.getRequestHeader().isSecure()) {
+				    count++;
+				} else {
+					raiseAlert(newRequest, "redirecttohttp");
+					return;
+				}
+			}
+			if(!newRequest.getRequestHeader().isSecure() && count == 10) { //For cases when even the 10th redirection is not https
```

Comment:
The expression "!newRequest.getRequestHeader().isSecure()" can be removed. At this point the request can only be secure, we raise an alert and return when it's HTTP.